x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/droppable/on_drop_handler.ts

```diff
@@ -124,9 +124,26 @@ function onFieldDrop(props: DropHandlerProps<DraggedField>, shouldAddField?: boo
   const layer = state.layers[target.layerId];
   const indexPattern = indexPatterns[layer.indexPatternId];
   const targetColumn = layer.columns[target.columnId];
+  // discourage already used operations for a field
+  const alreadyUsedOperations = new Set(
+    Object.values(layer.columns)
+      .map((column) =>
+        'sourceField' in column && column.sourceField === source.field.name
+          ? column.operationType
+          : undefined
+      )
+      .filter(Boolean) as string[]
```

Comment:
not: while it works right now, I would swap the operations order:
```suggestion
      .filter((column) =>
        'sourceField' in column && column.sourceField === source.field.name
      )
      .map(column => column.operationType)
```